geoportailv3/scripts/layers2es.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.paster import bootstrap
+import sys
+import getopt
+import json
+import requests
+from elasticsearch import helpers
+from elasticsearch.helpers import BulkIndexError
+from elasticsearch.exceptions import ConnectionTimeout
+from geoportailv3.lib.search import get_elasticsearch, get_index, ensure_index
+from pyramid.i18n import TranslationStringFactory, make_localizer
+from pyramid.interfaces import ITranslationDirectories
+from pyramid.request import Request
+
+
+"""
+Utility functions for importing layers metadata into Elasticsearch
+"""
+
+env = bootstrap('development.ini')
+from c2cgeoportal.models import DBSession, TreeItem, Role, LayerGroup
+from c2cgeoportal.views.entry import Entry
+
+
+def update_document(index, type, obj=None):
+    doc = {
+        "_index": index,
+        "_type": type,
+        "_id": str(obj['layer_id']) + "_" + obj['language']
+    }
+    doc['_source'] = {}
+    doc['_source']['role_id'] = obj['role_id']
+    doc['_source']['public'] = obj['public']
+    doc['_source']['language'] = obj['language']
+    doc['_source']['layer_id'] = obj['layer_id']
+    doc['_source']['name'] = obj['name']
+    doc['_source']['name_translated'] = obj['name_translated']
+    doc['_source']['description'] = obj['description']
+    doc['_source']['keywords'] = obj['keywords']
+    doc['_source']['metadata_name'] = obj['metadata_name']
+    return doc
+
+
+def statuslog(text):
+    sys.stdout.write(text)
+    sys.stdout.flush()
+
+
+def get_layers(theme):
+    layers = []
+
+    def flatten(x):
+        if 'children' in x:
+            flatten(x['children'])
+        elif type(x) is list:
+            for y in x:
+                flatten(y)
+        else:
+            layers.append({'id': x['id'], 'name': x['name']})
+    flatten(theme)
+    return layers
+
+
+def main():
+    try:
+        opts, args = getopt.getopt(sys.argv[1:], 'ri', ['reset', 'index'])
+    except getopt.GetoptError as err:
+        print str(err)
+        sys.exit(2)
+    index, reset = False, False
+    for o, a in opts:
+        if o in ('-r', '--reset'):
+            statuslog('\rResetting Index')
+            reset = True
+        if o in ('-i', '--index'):
+            statuslog('\rChecking Index')
+            index = True
+
+    request = env['request']
+    registry = env['registry']
+    ensure_index(get_elasticsearch(request), get_index(request), reset)
+
+    layers = []
+    roles = DBSession.query(Role).all()
+    for role in roles:
+        role_layers = []
+        req = Request.blank('/themes?interface=desktop&version=2&role_id='
+                            + str(role.id))
+        req.registry = request.registry
+        req.user = None
+        req.remote_addr = '127.0.0.1'
```

Comment:
@petzlux if the REMOTE_ADDR header is not set the role_id will be always None see here : https://github.com/camptocamp/c2cgeoportal/blob/233e68fbb130e82a8600725df86ffc06a5e645a5/c2cgeoportal/views/entry.py#L1179-L1180